### DISCUSSION TITLE
Differentiated control scheme: [ Keyboard | Mouse] controls [ individual | int | ext | all ] displays) + Differentiated menu item visibility

### DISCUSSION THREAD

**Discussion by porg** [MonitorControl/MonitorControl#1705] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues

### Is your feature request related to a problem? Please describe

I'd like to achieve this:

### a) Keyboard: ✅ Works 100% satisfying by combining:

```
Brightness and contrast:  [ Standard keyboard brightness keys                    ▾ ]
``` 

```
      Screen to control:  [ Depends on mouse pointer position                    ▾ ]
```

### b) Mouse: Via menu item: 💡 Okay but could be more differentiated

in https://github.com/MonitorControl/MonitorControl/pull/957#issuecomment-1047854270 @waydabber describes this possibility

- Settings  →  App menu
```
              Menu Icon:  [ Only if at least one slider is present               ▾ ]
```

```
     Brightness control:   ☐ Enable for Apple branded and built-in displays as well
```

- To achieve this:

| Situation | Outcome |
|---|---|
| Internal display only |  Menu Icon hidden |
| Internal display + external display |  Menu Icon shown |

### c) Menu item visibility: Current semantically indirect/unclean option "Only if at least one slider is present " does not allow a differentiated control scheme

- If `Menu Icon` gets 1-2 extra options:
  - `Only if at least two sliders are present` regardless of how many of those are internal or external
  - `Only if at least 1 external display is connected` as an even more concise option
- Then one could show the Menu Icon also with "1 external display only" when the MacBook is:
  - in clamshell mode
  - or has its internal display off so that the GPU can drive one more external display (possible e.g. with [BetterDisplay](https://github.com/waydabber/BetterDisplay) according to this [testimony](https://www.reddit.com/r/MacOS/comments/w9yas0/comment/kwb0p9o/)) )
- The settings would then look like this: 

```
              Menu Icon:  [ Only if at least 1 external display is connected     ▾ ]
```

```
     Brightness control:   ☐ Enable for Apple branded and built-in displays as well
```

### a) … keyboard

✅ Works very well already with:

```
Brightness and contrast:  [ Standard keyboard brightness keys                    ▾ ]
``` 

  - … many modifiers options exists:
> Use the brightness keys of your Apple keyboard to control brightness. Additionally you can hold:
>  - `  ⌃    `  to adjust the built-in display.
>  - `  ⌃   ⌘`  to adjust external displays.
>  - `  ⇧ ⌥  ` for fine control.
>  - `  ⌃ ⌥ ⌘` adjusts contrast on DDC compatible displays.

❗️  My particular desired modifier is missing, I'd love the following to be added:

>  - `⇧ ⌃ ⌥ ⌘` to adjust all displays.

### … b) mouse: Via menu item

✅ Works perfect with

  - Settings  →  App menu

```
      Multiple displays:  [ Use combined slider for all displays                 ▾ ]
``` 

❌ Except that this option is effectively unavailable when `Enable for Apple branded and built-in displays as well` is OFF which is needed for 1c. If 1bc do not have that quirky unclean/indirect option `Only if at least one slider is present` but instead the semantically cleaner options `Only if at least 1 external display is connected`  and `Only if at least two sliders are present` then you can have `Enable for Apple branded and built-in displays as well` is ON independently.

### Describe the solution you'd like

- 2a) `⇧ ⌃ ⌥ ⌘` to adjust all displays would be the minimal fix.

### Describe alternatives you've considered

- Fixing 1c) and subsequently thus decoupling 2b) would make the full 1abc 2ab suite possible! And would be the perfection.

**Comment by waydabber** [comment]
Maybe you can try switching to BetterDisplay as it has keyboard shortcut to control the built-in display only (Shift+Command+brightness keys) and all external displays only (Control+Command+brightness keys). The latter combo should be working with MonitorControl as well.

**Reply by porg** [reply]
Thanks for your reaction! And thanks for being so open-minded to recommend me an alternative software too. I will look into BetterDisplay and see if my use cases are better served there, and will share my experiences with you then.
